apis/apps/v1alpha1/type.go

```diff
@@ -651,3 +651,236 @@ type ConnectionCredential struct {
+// TODO: resources.
+// --------------------------------------------------------------------------------
+// |           |           |                      |  available in   |             |
+// | object    | attribute | var                  | template | Env  | description |
+// --------------------------------------------------------------------------------
+// | Namespace |           | KB_NAMESPACE         |          |      |             |
+// | Cluster   | name      | KB_CLUSTER_NAME      |          |      |             |
+// |           | UID       | KB_CLUSTER_UID       |          |      |             |
+// |           | comp name | KB_CLUSTER_COMP_NAME |          |      |             |
+// | Component | name      | KB_COMP_NAME         |          |      |             |
+// |           | replicas  | KB_COMP_REPLICAS     |          |      |             |
+// | Pod       | name      | KB_POD_NAME          |     x    |      |             |
+// |           | UID       | KB_POD_UID           |     x    |      |             |
+// |           | IP        | KB_POD_IP            |     x    |      |             |
+// |           | IPs       | KB_POD_IPS           |     x    |      |             |
+// |           | FQDN      | KB_POD_FQDN          |     x    |      |             |
+// |           | ordinal   | KB_POD_ORDINAL       |     x    |      |             |
+// | Host      | name      | KB_NODENAME          |     x    |      |             |
+// |           | IP        | KB_HOST_IP           |     x    |      |             |
+// | SA        | name      | KB_SA_NAME           |     x    |      |             |
+// --------------------------------------------------------------------------------
+
+// EnvVar represents a variable present in the env of Pod/Action or the template of config/script.
+type EnvVar struct {
+	// Name of the variable. Must be a C_IDENTIFIER.
+	// +required
+	Name string `json:"name"`
+
+	// Optional: no more than one of the following may be specified.
+
+	// Variable references $(VAR_NAME) are expanded using the previously defined variables in the current context.
+	// If a variable cannot be resolved, the reference in the input string will be unchanged.
+	// Double $$ are reduced to a single $, which allows for escaping the $(VAR_NAME) syntax: i.e.
+	// "$$(VAR_NAME)" will produce the string literal "$(VAR_NAME)".
+	// Escaped references will never be expanded, regardless of whether the variable exists or not.
+	// Defaults to "".
+	// +optional
+	Value string `json:"value,omitempty"`
+	// Source for the variable's value. Cannot be used if value is not empty.
+	// +optional
+	ValueFrom *EnvVarSource `json:"valueFrom,omitempty"`
+}
+
+// EnvVarSource represents a source for the value of an EnvVar.
+type EnvVarSource struct {
+	// Selects a key of a ConfigMap.
+	// +optional
+	ConfigMapKeyRef *corev1.ConfigMapKeySelector `json:"configMapKeyRef,omitempty"`
```

Comment:
What is the difference here with Pod.Container.EnvFrom.ConfigMapRef? Can it be removed?

Reply:
to be used to render the config & script templates